src/index.ts

```diff
@@ -44,31 +52,85 @@ import {
 
 import '../style/index.css';
 
+
+/**
+ * A class that tracks editor widgets.
+ */
+export
+interface IPDFJSTracker extends IInstanceTracker<PDFJSViewer> {}
+
+
+/* tslint:disable */
+/**
+ * The editor tracker token.
+ */
+export
+const IPDFJSTracker = new Token<IPDFJSTracker>('@jupyterlab/latex:IPDFJSTracker');
+/* tslint:enable */
+
+const latexPluginId = '@jupyterlab/latex:plugin';
+
 namespace CommandIDs {
-  export const openLatexPreview = 'latex:open-preview';
+  /**
+   * Open a live preview for a `.tex` document.
+   */
+  export
+  const openLatexPreview = 'latex:open-preview';
+
+  /**
+   * Reveal in the editor a position from the pdf using SyncTeX.
+   */
+  export
+  const synctexEdit = 'latex:synctex-edit';
+
+  /**
+   * Reveal in the pdf a position from the editor using SyncTeX.
+   */
+  export
+  const synctexView = 'latex:synctex-view';
 }
+
+/**
+ * The options for a SyncTeX view command,
+ * mapping the editor position the PDF.
+ */
+type ISynctexViewOptions = CodeEditor.IPosition;
+
+/**
+ * The options for a SyncTeX edit command,
+ * mapping the pdf position to an editor position.
+ */
+type ISynctexEditOptions = PDFJSViewer.IPosition;
+
 /**
  * The JupyterLab plugin for the LaTeX extension.
  */
 const latexPlugin: JupyterLabPlugin<void> = {
-  id: 'jupyterlab-latex:open',
-  requires: [IDocumentManager, IEditorTracker, ILayoutRestorer, IStateDB],
+  id: latexPluginId,
+  requires: [
+    IDocumentManager,
+    IEditorTracker,
+    ILayoutRestorer,
+    IPDFJSTracker,
+    ISettingRegistry,
+    IStateDB
+  ],
   activate: activateLatexPlugin,
   autoStart: true
 };
 
 /**
  * Make a request to the notebook server LaTeX endpoint.
  *
- * @param url - the path to the .tex file to watch.
+ * @param path - the path to the .tex file to watch.
  *
  * @param settings - the settings for the current notebook server.
  *
- * @returns a Promise resolved with the JSON response.
+ * @returns a Promise resolved with the text response.
  */
-export
-function latexRequest(url: string, settings: ServerConnection.ISettings): Promise<any> {
-  let fullUrl = URLExt.join(settings.baseUrl, 'latex', url);
+function latexBuildRequest(path: string, synctex: boolean, settings: ServerConnection.ISettings): Promise<any> {
+  let fullUrl = URLExt.join(settings.baseUrl, 'latex', 'build', path);
+  fullUrl += `?synctex=${synctex ? 1 : 0}`;
```

Comment:
What does this variable on build do differently? I thought you had separated the handlers and therefore the endpoints.

Reply:
Ah — I get it now. The synctex endpoint is for the search, this ensures that the search is possible by building the synctex doc in the first place.

Reply:
Exactly.